pkgs/development/libraries/mimalloc/default.nix

```
@@ -32,7 +33,11 @@ stdenv.mkDerivation rec {
   '';
 
   nativeBuildInputs = [ cmake ninja ];
-  cmakeFlags = [ "-DMI_INSTALL_TOPLEVEL=ON" ] ++ lib.optional secureBuild [ "-DMI_SECURE=ON" ];
+
+  cmakeFlags = [ "-DMI_INSTALL_TOPLEVEL=ON" ]
+    ++ lib.optional secureBuild "-DMI_SECURE=ON"
+    ++ lib.optional (overrideMalloc != null) "-DMI_OVERRIDE=${if overrideMalloc then "ON" else "OFF"}"
+    ++ lib.optionals stdenv.hostPlatform.isStatic [ "-DMI_BUILD_SHARED=OFF" "-DMI_BUILD_TESTS=OFF" ];
```

```suggestion
  cmakeFlags = [ "-DMI_INSTALL_TOPLEVEL=ON" "-DMI_OVERRIDE=${if overrideMalloc then "ON" else "OFF"}" ]
    ++ lib.optional secureBuild "-DMI_SECURE=ON"
    ++ lib.optionals stdenv.hostPlatform.isStatic [ "-DMI_BUILD_SHARED=OFF" "-DMI_BUILD_TESTS=OFF" ];
```

> instead of the null approach which falls back to whatever the upstream mimalloc does at any point in time.

I believe we'd like to be explicit with this in that case.